src/Interpreters/Access/InterpreterGrantQuery.cpp

```diff
@@ -178,6 +178,21 @@ namespace
                 elements_to_revoke.emplace_back(std::move(element_to_revoke));
         }
```

Comment:
Because there is no access element that can atomically represent `GRANT ALL ON *.*; REVOKE ALL ON mydb.*;`
Therefore, if we try to execute `REVOKE ALL ON *.*` from the user with the exact same rights, `element_to_revoke` will contain only `REVOKE ALL ON *.*` and the request will fail.

Reply:
Thank you for the explanation, I see it now.